### PR TITLE
fix(compiler): support more than 9 interpolations

### LINKED ISSUE #10253: angular2 Only support at most 9 interpolation values!
**I'm submitting a ...**  (check one with "x")

```
[ ] bug report
```

**Current behavior**
When I am trying to interpolate more than 9 properties of an object in template I am getting the following error:
Error: Uncaught (in promise): Template parse errors:
Only support at most 9 interpolation values!

**Expected/desired behavior**
It should be able to print any number of properties of an object

**Reproduction of the problem**
The issue could be reproduced at : 
http://plnkr.co/edit/3M7lw6U4RAuOPacM4rmj?p=preview

**What is the expected behavior?**
It should be able to interpolate any number of properties of an object

**What is the motivation / use case for changing the behavior?**
I am unable to print all details details of users

**Please tell us about your environment:**
- **Angular version:** 2.0.0-rc4
- **Browser:** all 
- **Language:** Typescript

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
Fixes #10253

### CODE REVIEW

File: modules/@angular/compiler/src/compiler_util/expression_converter.ts [hunk 1]
```diff
@@ -275,7 +275,12 @@ class _AstToIrVisitor implements cdAst.AstVisitor {
       args.push(this.visit(ast.expressions[i], _Mode.Expression));
     }
     args.push(o.literal(ast.strings[ast.strings.length - 1]));
-    return o.importExpr(resolveIdentifier(Identifiers.interpolate)).callFn(args);
+
+    return ast.expressions.length <= 9 ?
```
Comment (DzmitryShylovich) [inline modules/@angular/compiler/src/compiler_util/expression_converter.ts]: shouldn't u use https://github.com/angular/angular/blob/2e78b76fcfe03fa9dc043f4395fb8ed8cb4e8736/modules/@angular/core/src/linker/view_utils.ts#L48 here?

File: modules/@angular/compiler/src/compiler_util/expression_converter.ts [hunk 2]
```diff
@@ -275,7 +275,12 @@ class _AstToIrVisitor implements cdAst.AstVisitor {
...
+        o.importExpr(resolveIdentifier(Identifiers.interpolate)).callFn(args) :
```
Comment (vicb) [inline modules/@angular/compiler/src/compiler_util/expression_converter.ts]: `inlineInterpolate` and `interpolate` sounds better IMO

File: modules/@angular/core/src/linker/view_utils.ts [hunk 3]
```diff
@@ -45,7 +45,13 @@ export function addToArray(e: any, array: any[]) {
   array.push(e);
 }
 
-export const MAX_INTERPOLATION_VALUES = 9;
+export function interpolateMany(valueCount: number, constAndInterp: string[]) {
+  let result = '';
+  for (let i = 0; i < valueCount * 2; i = i + 2) {
+    result = result + constAndInterp[i] + constAndInterp[i + 1];
```
Comment (vicb) [inline modules/@angular/core/src/linker/view_utils.ts]: any test for `null` / `undefined` ?

Comment (pkozlowski-opensource) [inline modules/@angular/core/src/linker/view_utils.ts]: Oh, good catch!!! Will change + add a test. Thnx!

File: modules/@angular/core/test/linker/integration_spec.ts [hunk 4]
```diff
@@ -61,6 +61,17 @@ function declareTests({useJit}: {useJit: boolean}) {
         expect(fixture.nativeElement).toHaveText('');
       });
 
+      it('should support more than 9 interpolations in one text node', () => {
```
Comment (vicb) [inline modules/@angular/core/test/linker/integration_spec.ts]: - don't like the magic "9" here, could we say "an arbitrary number" and generate 30 interp ?
- "in one text node" is not what we test - we test in "an element"_(Tobias wants test splitting across several text nodes)_

`const fixture = TestBed.overrideComponent(MyComp, {set: {template}}).createComponent(MyComp);`

File: modules/@angular/compiler/src/compiler_util/expression_converter.ts (hunk 2 above)
Comment (pkozlowski-opensource) [inline modules/@angular/compiler/src/compiler_util/expression_converter.ts]: I don't mind either way, can change.

File: modules/@angular/core/test/linker/integration_spec.ts [hunk 5]
```diff
@@ -61,6 +61,17 @@ function declareTests({useJit}: {useJit: boolean}) {
         expect(fixture.nativeElement).toHaveText('');
       });
 
+      it('should support an arbitrary an arbitrary number of interpolations in an element', () => {
```
Comment (alfaproject) [inline modules/@angular/core/test/linker/integration_spec.ts]: `an arbitrary` x2

**Comment by pkozlowski-opensource** [comment]
@vicb @tbosch I believe that I've addressed all the review comments. Please have another look or just stick the merge label :-)

**Pull request merged**